src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -14,9 +14,11 @@ public partial class SocketAsyncEventArgs : EventArgs, IDisposable
         // Buffer,Offset,Count property variables.
         private WSABuffer _wsaBuffer;
         private IntPtr _ptrSingleBuffer;
+        private GCHandle _singleBufferGCHandle;
 
         // BufferList property variables.
         private WSABuffer[] _wsaBufferArray;
+        private GCHandle[] _multipleBufferGCHandles;
```

Comment:
We're already concerned about the size of SocketAsyncEventArgs.  I wonder if it would be better to only have a _bufferGCHandles, regardless of whether a single or multiple buffers are being used, and if single, just use _bufferGCHandles[0]?  The downside is worst case you'll allocate a 1-length array, and that worst case may be very common.